### ISSUE TITLE
latest oc docker image update throwing 'illegal instruction' errors

### ISSUE THREAD

**Issue by BackedUpBooty** [owncloud/core#40929]
This is on a Synology DS920+ which has had a working instance of Owncloud (with MariaDB and Redis) for the past 2 years. Please note that the method below which results in errors in 10.12 works fine for 10.12.1.

### Steps to reproduce
1. Install Owncloud with tag owncloud/server:10.12
2. Redis tag redis:6
3. MariaDB tag mariadb:10.11

### Expected behaviour
Owncloud starts up normally and is able to connect to the database

### Actual behaviour
Ownlcloud container does not start, it loops the error
```
/usr/bin/occ: line 16:    63 Illegal instruction     (core dumped) setpriv --reuid=www-data --regid=www-data --init-groups php /var/www/owncloud/occ "$@"
``` 
after attempting to write the config file, and no config file actually gets written into the `/mnt/data` mapped volume (not a permissions error, permissions were set wide open for this just to check). 

### Server configuration
**Operating system**: Synology DSM 7.2, Docker, docker-compose.yml is here https://lenpaste.pointtosource.com/TzqzOBkj

**Web server:** N/A

**Database:** MariaDB 10.11

**PHP version:** 

**ownCloud version:** (see ownCloud admin page)
Cannot access owncloud anymore 

**Updated from an older ownCloud or fresh install:**
Both. This was noticed after today's image update, and has the same result on a fresh install with nothing in the mapped folders. 

**Where did you install ownCloud from:**
owncloud/server

**Signing status (ownCloud 9.0 and above):**
??

```
Login as admin user into your ownCloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results into https://gist.github.com/ and puth the link here.
```
Cannot

**The content of config/config.php:**
```
Log in to the web-UI with an administrator account and click on
'admin' -> 'Generate Config Report' -> 'Download ownCloud config report'
This report includes the config.php settings, the list of activated apps
and other details in a well sanitized form.

or 

If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your ownCloud installation folder

*ATTENTION:* Do not post your config.php file in public as is. Please use one of the above
methods whenever possible. Both, the generated reports from the web-ui and from occ config:list
consistently remove sensitive data. You still may want to review the report before sending.
If done manually then it is critical for your own privacy to dilligently
remove *all* host names, passwords, usernames, salts and other credentials before posting.
You should assume that attackers find such information and will use them against your systems.
```
This doesn't get created as the container seems to have an issue early on, however the `overwrite.config.php` does, link to that here https://lenpaste.pointtosource.com/etX9jGgp

**List of activated apps:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your ownCloud installation folder.
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
No
**Are you using encryption:** yes/no
N/A
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
No

### Client configuration
**Browser:**
Chrome
**Operating system:**
Win11

#### Web server error log
```
Insert your webserver log here
```

#### ownCloud log (data/owncloud.log)
```
Insert your ownCloud log here
```
Container logs loop as follows:
```
Creating volume folders...
Creating hook folders...
Waiting for MySQL...
services are ready!
Waiting for Redis...
services are ready!
Writing config file...
Fixing base perms...
Fixing data perms...
Fixing hook perms...
/usr/bin/occ: line 16:    60 Illegal instruction     (core dumped) setpriv --reuid=www-data --regid=www-data --init-groups php /var/www/owncloud/occ "$@"
Upgrading server database...
/usr/bin/occ: line 16:    67 Illegal instruction     (core dumped) setpriv --reuid=www-data --regid=www-data --init-groups php /var/www/owncloud/occ "$@"
```

**Comment by tillo** [comment]
I believe that the patch https://github.com/owncloud-docker/base/commit/0ea9aaef59ea2b8c80508a719a02197621562dca introduced this issue as it doesn't seem that setpriv can run on a standard unprivileged container.

**Comment by tillo** [comment]
For those wondering, doing a rollback to owncloud/server:10.12.1 is not an option, because downgrades of patch version surprisingly throws an unhandled exception.
```
owncloud  | Writing objectstore config...
owncloud  | Writing php config...
owncloud  | Updating htaccess config...
owncloud  | An unhandled exception has been thrown:
owncloud  | OC\HintException: [0]: Downgrading is not supported and is likely to cause unpredictable issues (from 10.12.2.1 to 10.12.1.3) ()
```

**Comment by tillo** [comment]
I confirm that /usr/bin/php7.4 crashes in the latest owncloud/php image as well.
```
root@omissis:~# docker pull owncloud/php
Using default tag: latest
latest: Pulling from owncloud/php
Digest: sha256:bf06ede4767e48a1888cc3ba5134ef94da2c04094979ed6d9afcff167c5fba5a
Status: Image is up to date for owncloud/php:latest
docker.io/owncloud/php:latest
root@omissis:~# docker run -it owncloud/php bash
root@92971048fd75:/var/www/html# /usr/bin/php7.4
Illegal instruction (core dumped)
```

**Comment by tillo** [comment]
The plot thickens: even by reinstalling the package php7.4-cli (7.4.3-4ubuntu2.19) from the Ubuntu repositories, it keeps crashing. Currently investigating the image libraries and runtime.

**Comment by tillo** [comment]
Confirmed: disabling imagick works.

```
# phpdismod imagick && /usr/bin/php7.4 -h 2>&1 |head
Usage: php7.4 [options] [-f] <file> [--] [args...]
   php7.4 [options] -r <code> [--] [args...]
   php7.4 [options] [-B <begin_code>] -R <code> [-E <end_code>] [--] [args...]
   php7.4 [options] [-B <begin_code>] -F <file> [-E <end_code>] [--] [args...]
   php7.4 [options] -S <addr>:<port> [-t docroot] [router]
   php7.4 [options] -- [args...]
   php7.4 [options] -a

  -a               Run interactively
  -c <path>|<file> Look for php.ini file in this directory
```

Seems like using pecl to compile imagick during image creation wasn't the right call, in the meantime everyone can just disable the module in the server image to make it work.

I'm using the following in docker compose.

```
        command: |
            /bin/sh +x -c ' \
                phpdismod imagick \
                && /usr/bin/owncloud server'
```

**Comment by KillerTic** [comment]
Can confirm, that it works for me. Container is up and running and clients connected again.
Thanks for the quick interims fix!

**Comment by phil-davis** [comment]
@jnweiger see above ^
There seems to be some issue with imagick in the PHP docker.

**Comment by BackedUpBooty** [comment]
> ```
>         command: |
>             /bin/sh +x -c ' \
>                 phpdismod imagick \
>                 && /usr/bin/owncloud server'
> ```

Legendary, good stuff on such a quick patch. Thank you!

**Comment by SigHunter** [comment]
I can also see the following messages in dmesg:
`[27355.310284] traps: php[337995] trap invalid opcode ip:7fb1021d4a9c sp:7ffe6f079960 error:0 in libMagickCore-7.Q16HDRI.so.10.0.1[7fb1021c8000+284000]
[27355.703629] traps: php[338008] trap invalid opcode ip:7f9a1ea5fa9c sp:7ffebfb457f0 error:0 in libMagickCore-7.Q16HDRI.so.10.0.1[7f9a1ea53000+284000]
`

**Comment by jnweiger** [comment]
Ouch. we wanted to use a selfcompiled libmagic with upcoming 10.13.0 docker to add support for heic image format.
Unfortunately this automatically "upgraded" all existing 10.11 and 10.12 docker images too, instead of only the current 10.13.0-beta.1 image - sorry for this collateral damage.

Thanks for finding this and diagnosing it as far as pointing to `libMagickCore-7.Q16HDRI.so.10.0.1`
That is very helpful!

There must be some very silly compiler optimization flags enabled.

**Comment by jnweiger** [comment]
Indeed. There is an absolutely silly `gcc -march=native` involved. That is the optimizer strategy that uses every available feature of **this current** CPU for best performance.
I have a test build with `gcc -mtune=generic` for building this imagemagic thing. Coud someone with an affected CPU here please try this:
`docker run --rm  jnweiger/owncloud-php-test:20230819 php -v`

**Comment by noc-noc** [comment]
> Could someone with an affected CPU here please try this:
> `docker run --rm jnweiger/owncloud-php-test:20230819 php -v`

Output is:
```
PHP 7.4.3-4ubuntu2.19 (cli) (built: Jun 27 2023 15:49:59) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.3-4ubuntu2.19, Copyright (c), by Zend Technologies
```

**Comment by jnweiger** [comment]
Thanks for testing! No more illegal instructions!
I'll submit that patch as an official fix. As soon as it gets merged, you should have fixed 10.11 and 10.12 docker images again.
